model/symflower/symflower.go

```diff
@@ -12,18 +12,23 @@ import (
 	"github.com/symflower/eval-dev-quality/util"
 )
 
-// ModelSymflower holds a Symflower model using the locally installed CLI.
-type ModelSymflower struct{}
+// Model holds a Symflower model using the locally installed CLI.
+type Model struct{}
 
-var _ model.Model = (*ModelSymflower)(nil)
+// NewModel returns a Symflower model.
+func NewModel() (model model.Model) {
+	return &Model{}
+}
+
+var _ model.Model = (*Model)(nil)
```

Comment:
no real need to keep that check here since the function above contains the same type scenario (trying to return a model instance as a `model.Model`.

similar for a lot of other cases in this commit

Reply:
Its a pattern for me that we do internally as well: explicitly state the interface you want to implement. then put the methods afterwards.